### PR TITLE
[STT-1569] Preserve custom tag styles when pasting between editors

### PR DESCRIPTION
**Pull request by eos87**
### Description
When copying text between two editor3 instances (including across separate browser windows), inline styles coming from custom editor tags were lost on paste. 

### What was the problem:
When pasting from another window, editor3 receives raw HTML from the browser clipboard. Custom tag styles are represented as CSS (e.g. `border-block-end: 4px double ...`) in that HTML. The parser had no way to recognize those CSS values and convert them back into the internal DraftJS inline style names, so the styles were silently dropped.

### What has changed:
- The HTML parser now recognizes pasted CSS matching any configured custom tag style and converts it back to the corresponding DraftJS inline style. Both raw CSS variable values (as Firefox preserves them) and browser-computed values (as Chrome resolves them) are handled.
- Because DraftJS's `convertFromHTML` strips inline styles, the recognized spans are first converted to [PUA](https://en.wikipedia.org/wiki/Private_Use_Areas) Unicode marker characters that survive the conversion, then the markers are stripped and the inline styles are applied to the correct character ranges in post-processing.
- The cross-window paste identity check now validates that the clipboard text hash matches the stored copy metadata, preventing stale metadata from interfering.
- Tests cover: the full CSS→inline style pipeline, `extractCustomTagRanges` boundary cases, `editor3StateToHtml` custom tag emission, and the `pasteContentFromOpenEditor` paste path including style preservation.

Resolves: [STT-1569]

[STT-1569]: https://sofab.atlassian.net/browse/STT-1569?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/core/editor3/components/handlePastedText.ts [hunk 1]
```diff
@@ -23,41 +24,80 @@ function removeMediaFromHtml(htmlString): string {
...
-            }
-        }
+    if (internalClipboard == null) {
+        return 'not-handled';
     }
 
-    return 'not-handled';
+    const blocksArray = [];
+
+    internalClipboard.forEach((b) => blocksArray.push(b));
```
Comment (BrianMwangi21) [inline scripts/core/editor3/components/handlePastedText.ts]: Can this be like `const blocksArray = Array.from(internalClipboard);` ?

File: scripts/core/editor3/html/from-html/index.ts [hunk 2]
```diff
@@ -8,8 +8,125 @@ import {
...
+    document.body.appendChild(probe);
+
+    for (const [styleName, cssProps] of Object.entries(customEditorTagStyleMap)) {
+        for (const [propName, rawValue] of Object.entries(cssProps)) {
+            if (!map.has(propName)) {
+                map.set(propName, new Map());
+            }
+
+            const valueMap = map.get(propName);
+            const rawLower = (rawValue as string).toLowerCase();
```
Comment (BrianMwangi21) [inline scripts/core/editor3/html/from-html/index.ts:65]: Suggestion: `const rawLower = String(rawValue).toLowerCase();`

**Review by petrjasek — APPROVED** [review]

File: scripts/core/editor3/html/from-html/index.ts (hunk 2 above)
Comment (eos87) [reply scripts/core/editor3/html/from-html/index.ts:65]: I rather keep it this way. If a value in that map is ever nullish, it's a bug in the map definition itself, not something to silently swallow at runtime.

File: scripts/core/editor3/components/handlePastedText.ts (hunk 1 above)
Comment (eos87) [reply scripts/core/editor3/components/handlePastedText.ts]: done.

**Review by BrianMwangi21 — APPROVED** [review]
